src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -222,6 +222,22 @@ public static ImmutableArray<byte> ToImmutable(this MemoryStream stream)
             return builder.ToImmutableAndFree();
         }
 
+
+        /// <summary>
+        /// Maps an immutable array through a function that returns ValueTasks, returning the new ImmutableArray.
```

Comment:
Consider documenting that the passed function should not depend on the current threading context.

Reply:
@333fred Not sure I follow. Can you clarify?

Reply:
You're doing a look and calling `ConfigureAwait(false)` on the function. If it was assuming that it would start from the same threading context every loop, it would break.

Reply:
I'll do this as a follow-up to this PR so this can meet the impending snap.